addons/ApplicationInsights/Tweek.Addons.ApplicationInsights/ApplicationInsightAddon.cs

```diff
@@ -1,15 +1,38 @@
 ﻿using System;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
 using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Tweek.ApiService.Addons;
 
 namespace Tweek.Addons.ApplicationInsights
 {
+    public class ApiClientTelemetryInitializer : ITelemetryInitializer
+    {
+        IHttpContextAccessor httpContextAccessor;
+
+        public ApiClientTelemetryInitializer(IHttpContextAccessor httpContextAccessor)
+        {
+            this.httpContextAccessor = httpContextAccessor;
+        }
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (httpContextAccessor.HttpContext == null) return;
+            if (httpContextAccessor.HttpContext.Request.Headers.TryGetValue("x-api-client", out var values)){
```

Comment:
I think HttpContext is created per request, so I don't think Request can be null